src/cron/personalizedDigest.ts

```diff
@@ -0,0 +1,35 @@
+import { notifyGeneratePersonalizedDigest } from '../common';
+import { UserPersonalizedDigest } from '../entity/UserPersonalizedDigest';
+import { Cron } from './cron';
+
+const cron: Cron = {
+  name: 'personalized-digest',
+  handler: async (con, logger) => {
+    const personalizedDigestQuery = con
+      .createQueryBuilder()
+      .from(UserPersonalizedDigest, 'upd');
+
+    const personalizedDigestStream = await personalizedDigestQuery.stream();
+
+    personalizedDigestStream.on(
+      'data',
+      (personalizedDigest: UserPersonalizedDigest) => {
+        notifyGeneratePersonalizedDigest(logger, personalizedDigest);
+      },
+    );
+
+    await new Promise((resolve, reject) => {
+      personalizedDigestStream.on('error', (error) => {
+        logger.error(
+          { err: error },
+          'streaming personalized digest subscriptions failed',
+        );
+
+        reject(error);
+      });
```

Comment:
If stream fails for any reason would cron retry? Just wanna make sure that we don't get the flow:
- we start to schedule digest generation
- it fails mid generation (but some digest are already scheduled)
- it retries and schedules more

Reply:
Use `redis` or api db storage to store data on which digests were sent and when. As a mean for deduplication.

Reply:
That makes sense, though not sure if our cron jobs retry when failing. As as I know they don't.

Reply:
Cron job do have a retry mechanism. They actually also have a timeout mechanism so we need to make sure it doesn't take too long

Reply:
What is the timeout?

Reply:
Right now it's 5min but I think you can adjust it in pulumi

Reply:
I think 5 minutes should be plenty, but will monitor with initial deployment.

Reply:
Oh, nice. Good to know it was brought up, now I know it retries.